flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java

```diff
@@ -140,7 +140,8 @@
 
 	public static final ConfigOption<Long> SLOT_IDLE_TIMEOUT =
 		key("slot.idle.timeout")
-			.defaultValue(10L * 1000L)
+			// default matches heartbeat.timeout so that sticky allocation is not lost on timeouts for local recovery
+			.defaultValue(HeartbeatManagerOptions.HEARTBEAT_TIMEOUT.defaultValue())
```

Comment:
reminder: if this results in a different default value, I would suggest to make a note of the change in the "Release Notes" field of the JIRA ticket.